src/components/Report/Report.tsx

```diff
@@ -0,0 +1,472 @@
+}: DataTablePaginationProps<TData>) {
+  return (
+    <div className="flex items-center justify-between px-2">
+      <div className="flex-1 text-sm text-muted-foreground">
+        {table.getFilteredSelectedRowModel().rows.length} of{' '}
+        {table.getFilteredRowModel().rows.length} row(s) selected.
+      </div>
+      <div className="flex items-center space-x-6 lg:space-x-8">
+        <div className="flex items-center space-x-2">
+          <p className="text-sm font-medium">Rows per page</p>
+        </div>
+        <div className="flex w-[100px] items-center justify-center text-sm font-medium">
+          Page {table.getState().pagination.pageIndex + 1} of{' '}
+          {table.getPageCount()}
+        </div>
+        <div className="flex items-center space-x-2">
+          <Button
+            variant="outline"
+            className="hidden h-8 w-8 p-0 lg:flex"
+            onClick={() => table.setPageIndex(0)}
+            disabled={!table.getCanPreviousPage()}
+          >
+            <span className="sr-only">Go to first page</span>
+            <DoubleArrowLeftIcon className="h-4 w-4" />
+          </Button>
+          <Button
+            variant="outline"
+            className="h-8 w-8 p-0"
+            onClick={() => table.previousPage()}
+            disabled={!table.getCanPreviousPage()}
+          >
+            <span className="sr-only">Go to previous page</span>
+            <ChevronLeftIcon className="h-4 w-4" />
+          </Button>
+          <Button
+            variant="outline"
+            className="h-8 w-8 p-0"
+            onClick={() => table.nextPage()}
+            disabled={!table.getCanNextPage()}
+          >
+            <span className="sr-only">Go to next page</span>
+            <ChevronRightIcon className="h-4 w-4" />
+          </Button>
+          <Button
+            variant="outline"
+            className="hidden h-8 w-8 p-0 lg:flex"
+            onClick={() => table.setPageIndex(table.getPageCount() - 1)}
+            disabled={!table.getCanNextPage()}
+          >
+            <span className="sr-only">Go to last page</span>
+            <DoubleArrowRightIcon className="h-4 w-4" />
+          </Button>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+const ReportColumns = () => {}
+
+const SearchInput = ({
+  placeholder,
+  onChange,
+  value,
+}: {
+  placeholder: string
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
+  value: string
+}) => {
+  return (
+    <div className="relative flex w-full max-w-sm items-center space-x-2">
+      <Input
+        type="search"
+        value={value}
+        onChange={onChange}
+        placeholder={placeholder}
+      />
+      <Button
+        type="button"
+        variant={'outline'}
+        size={'icon'}
+        className="absolute right-0 top-0"
+        onClick={() => {
+          /* Optional: Handle search button click */
+        }}
+      >
+        <Search size={15} />
+      </Button>
+    </div>
+  )
+}
```

Comment:
There is a lot going on in this file. Please can you put the child components into different files

Reply:
yes

Reply:
will make new pull request

Reply:
With this same issue or another issue?

Reply:
same